virtool/subtractions/data.py

```diff
@@ -83,33 +85,84 @@ async def find(self, find: str, short: bool, ready: bool, query: MultiDictProxy)
-            db_query,
-            query,
-            base_query={"deleted": False},
-            sort="name",
-            projection=PROJECTION,
-        )
+        data = await self._mongo.subtraction.aggregate(
+            [
+                {
+                    "$facet": {
+                        "total_count": [
+                            {"$count": "total_count"},
+                        ],
+                        "found_count": [
+                            {"$count": "found_count"},
+                        ],
+                        "page_count": [
+                            {"$count": "page_count"},
+                        ],
+                        "per_page": [
+                            {"$count": "per_page"},
+                        ],
+                        "page": [
+                            {"$count": "page"},
+                        ],
+                        "documents": [
+                            {
+                                "$match": query,
+                            },
+                            {"$sort": {"name": 1}},
+                            *lookup_nested_user_by_id(local_field="user.id"),
+                            *lookup_minimal_job_by_id(),
+                        ],
+                    }
+                },
+                {
+                    "$project": {
+                        "documents": {
+                            "_id": True,
+                            "count": True,
+                            "created_at": True,
+                            "file": True,
+                            "ready": True,
+                            "job": True,
+                            "name": True,
+                            "nickname": True,
+                            "user": True,
+                        },
+                        "total_count": {
+                            "$arrayElemAt": ["$total_count.total_count", 0]
+                        },
+                        "found_count": {
+                            "$arrayElemAt": ["$found_count.found_count", 0]
+                        },
+                        "page_count": {
+                            "$arrayElemAt": ["$page_count.page_count", 0]
+                        },
+                        "per_page": {
+                            "$arrayElemAt": ["$per_page.per_page", 0]
+                        },
+                        "page": {
+                            "$arrayElemAt": ["$page.page", 0]
+                        },
+                    }
+                },
+            ]
+        ).to_list(length=1)
 
-        documents, ready_count = await asyncio.gather(
-            apply_transforms(
-                data["documents"],
-                [AttachUserTransform(self._mongo, ignore_errors=True)],
-            ),
-            self._mongo.subtraction.count_documents({"ready": True}),
-        )
+        data = data[0]
+
+        documents = data["documents"]
```

Comment:
Check behaviour when no subtraction matching the search query are found. Also if you are only returning the result there is no need to assign it to variables before returning